messaging/src/main/java/org/axonframework/eventhandling/AnnotationEventHandlerAdapter.java

```diff
@@ -105,6 +105,7 @@ public boolean canHandle(EventMessage<?> event) {
     @Override
     public boolean canHandleType(Class<?> payloadType) {
         return inspector.getHandlers(listenerType)
+                        .filter(messageHandler -> !messageHandler.hasAnnotation(ResetHandler.class))
```

Comment:
The idea of these meta-annotations is that we shouldn't check on the actual annotation used. Instead, we should check whether the MessageHandler annotation declares a type of message that is compatible with the EventMessage.

Reply:
Good point. Resolved just now.